### PR TITLE
Add simple cache to access witness

### PR DESCRIPTION
**Pull request by acolytec3**
WIP

### CODE REVIEW

File: packages/vm/src/runBlock.ts [hunk 1]
```diff
@@ -438,6 +438,8 @@ async function applyBlock(vm: VM, block: Block, opts: RunBlockOpts): Promise<App
     await accumulateParentBlockHash(vm, block.header.number, block.header.parentHash)
   }
 
+  vm.evm.systemVerkleAccessWitness?.flushCache()
```
Comment (gabrocheleau) [inline packages/vm/src/runBlock.ts:441]: Out of correctness, could we make that part of a isActivatedEIP(6800) conditional?

File: packages/vm/src/runTx.ts [hunk 2]
```diff
@@ -802,6 +802,7 @@ async function _runTx(vm: VM, opts: RunTxOpts): Promise<RunTxResult> {
     )
   }
 
+  vm.evm.verkleAccessWitness?.flushCache()
```
Comment (gabrocheleau) [inline packages/vm/src/runTx.ts:805]: Same as above (part of `isActivatedEIP(6800)` )

File: packages/evm/test/verkle.spec.ts [hunk 3]
```diff
@@ -42,4 +42,68 @@ describe('verkle tests', () => {
     const retrievedValue = await sm.getStorage(address, setLengthLeft(bigIntToBytes(2n), 32))
     assert.deepEqual(retrievedValue, setLengthLeft(bigIntToBytes(1n), 32))
   })
+  it('should revert and access witness should not contain a write access due to OOG', async () => {
```
Comment (gabrocheleau) [inline packages/evm/test/verkle.spec.ts:45]: Nice tests, simple and to-the-point.

File: packages/evm/src/evm.ts [hunk 4]
```diff
@@ -632,6 +642,7 @@ export class EVM implements EVMInterface {
           }
```
Comment (gabrocheleau) [inline packages/evm/src/evm.ts:642]: Should we also clearCaches when the code size exceeds maximum code size?

File: packages/evm/src/evm.ts [hunk 5]
```diff
@@ -663,6 +676,7 @@ export class EVM implements EVMInterface {
       )
       gasLimit -= createCompleteAccessGas
       if (gasLimit < BIGINT_0) {
+        message.accessWitness?.clearCache()
         if (this.DEBUG) {
           debug(
             `ContractCreateComplete access gas (${createCompleteAccessGas}) caused OOG (-> ${gasLimit})`,
```
Comment (gabrocheleau) [inline packages/evm/src/evm.ts:682]: Should we also add a corresponding flushCaches in the else statement for contractCreateComplete?

**Review by gabrocheleau — APPROVED** [review]
Looks great! Thanks a lot for tackling this OOG stuff. 
Left a few comments, can be merged once addressed

File: packages/evm/src/evm.ts (hunk 4 above)
Comment (g11tech) [reply packages/evm/src/evm.ts:642]: we haven't written any code chunks here, so no need

**Pull request merged**